core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt

```diff
@@ -130,4 +157,29 @@ class FileResponseHandler(
             storage.removeEventCallback(it.groupValues[1])
         }
     }
+
+    private fun triggerBackOff(withSizeUpdate: Boolean = false) {
+        logger?.debug("back off to retry sending events later.")
+        backoff.set(true)
+        if (retries.incrementAndGet() <= configuration.flushMaxRetries) {
+            currentFlushInterval *= 2
+            if (withSizeUpdate) {
+                currentFlushQueueSize = (currentFlushQueueSize * 2).coerceAtMost(maxQueueSize)
+            }
+        } else {
+            // stop scheduling new calls since max retries exceeded
+            eventPipeline.exceededRetries = true
+            logger?.debug("Max retries ${configuration.flushMaxRetries} exceeded, temporarily stop scheduling new events sending out.")
```

Comment:
while we stop sending for all traffic, should we recover it sometime later? Or just when next time user opens the app again? @justin-fiedler @yuhao900914  any thought?

Reply:
I think we should recover it sometime later. 
I feel there has some complaints about the data latency, we should recover it as soon as possible.

Reply:
I agree that we should have this recover at some point. In that case we should probably make it be configurable by the end user e.g. something like `maxBackoffInterval`.

Reply:
I added a recover after, feel free to take a look. Not sure if we another configuration is easier for people to know or better to config it on our end.